### ISSUE TITLE
I2C communication failure: (iokit/?) unknown subsystem error

### ISSUE THREAD

**Issue by blindflugmoritz** [waydabber/m1ddc#33]
It worked perfectly until the last week. 

Now when i send the command
m1ddc set input 17

I always get:
I2C communication failure: (iokit/?) unknown subsystem error

I can do other commands but i cant run the set input command somehow. :(

any idea how to debug? 

thanks & best M

**Comment by PrinceJayJiao** [comment]
> Getting the same. Are you using a hub? I think it might be related to that.

I use a hub too and get the same question. Have you find a solution? thanks

**Comment by PrinceJayJiao** [comment]
> > Getting the same. Are you using a hub? I think it might be related to that.
> 
> I use a hub too and get the similar question. Have you find a solution? thanks this is my err info DDC communication failure: (iokit/?) unknown subsystem error

I asked the hub seller. They say that the hub do not support ddc. My hub brand is ugreen. What an upset news! it really take me so long time.

**Comment by kevintosh** [comment]
Hi! :) It's late for this post, but I didn't want to open a new thread/issue for this, so just an update here.

My current config:
- MacBook Air M4
- Apple Cinema Display 30" (yes, I know, they are over 20 years apart 😂)
- Belkin Thunderbolt Express Dock
- HP USB-C Dock G5

I was breaking my head why "suddenly" MonitorControl (BetterDisplay, Lunar, m1ddc, and everything) stopped working... when I realized that "suddenly" was actually me switching from the Belkin Dock to the HP one.

Apparently the Belkin Dock does proper DDC Relay through the Thunderbolt port (even if I have it connected through a Thunderbolt 3/2 Adapter from Apple) and the HP one does not.

To be fair, the Mac goes through USB-C to the Dock, a DisplayPort to Mini DisplayPort Adapter, and then to the Cinema, so it could even be the adapter (but I doubt it, looks passive) and yes, I'm trying to marry 20-year apart techs.

So, tl;dr:

`DDC communication failure: (iokit/?) unknown subsystem error`

happens only on the HP USB Dock, which also strangely, causes MonitorControl to crash (I'm not sure if same error, or unrelated).

On the Belkin everything _seems_ to be working dandy right now...
